startup.py

```diff
@@ -241,4 +241,64 @@ def prepare_launch(self, exec_path, args, file_to_open=None):
 
         :returns: :class:`LaunchInformation` instance
         """
-        return LaunchInformation(exec_path, args)
\ No newline at end of file
+
+        # Get the basic environment variables to launch Nuke. Do this by reusing Toolkit classic's
+        # bootstrap code.
+        startup_python_path = os.path.join(
+            self.disk_location,
+            "python",
+            "startup"
+        )
+
+        with self.temporary_import(startup_python_path, "bootstrap") as bootstrap:
+            launch_plugins = self.get_setting("launch_builtin_plugins")
+
+            if launch_plugins:
+                self.logger.info("Launch plugins: %s", launch_plugins)
+
+                # Get Nuke environment for plugin launch.
+                required_env, required_args = bootstrap.get_plugin_startup_env(
+                    launch_plugins, exec_path, args, file_to_open
+                )
+
+                # Add std context and site info to the env.
+                required_env.update(self.get_standard_plugin_environment())
+
+                # Make sure we are picking the right engine.
+                required_env["SHOTGUN_ENGINE"] = self.engine_name
+            else:
+                self.logger.info("Preparing Nuke Launch via Toolkit Classic methodology ...")
+
+                # Get Nuke environment for Toolkit Classic launch.
+                required_env, required_args = bootstrap.get_classic_startup_env(
+                    exec_path, args, file_to_open
+                )
+                # Add context information info to the env.
+                required_env["TANK_CONTEXT"] = sgtk.Context.serialize(self.context)
+                required_env["TANK_ENGINE"] = self.engine_name
+
+            self.logger.debug("Launch environment: %s", pprint.pformat(required_env))
+            self.logger.debug("Launch arguments: %s", required_args)
+
+        return LaunchInformation(exec_path, required_args, required_env)
+
+    @contextlib.contextmanager
+    def temporary_import(self, module_path, module_name):
```

Comment:
Useful pattern to import a module temporarily. Most engines probably need this. Maybe this needs to be refactored?